### PR TITLE
Compress Embeddings with multiple axes to 8 bit

### PR DESCRIPTION
**Pull request by l-bat**
### Reason for changes

To further reduce the model footprint

### Related tickets

124822

### Tests

test_quantize_Gather_with_multiple_reduction_axes_in_8bit
test_not_quantize_Gather_with_multiple_reduction_axes_if_all_layers

### CODE REVIEW

**Comment by ljaljushkin** [comment]
Have you succeeded with accuracy validation on chatglm3 model with your PR and before it?
Do you see any changes in accuracy? Could you share the results?

File: tests/openvino/native/quantization/test_weights_compression.py [hunk 1]
```diff
@@ -355,12 +355,22 @@ def test_data_based_criterion(mode, ref_scores, ref_act_scores, mocker):
 
 
 @pytest.mark.parametrize("mode", (CompressWeightsMode.INT8_SYM, CompressWeightsMode.INT8_ASYM))
-def test_not_quantize_with_multiple_reduction_axes(mode):
+def test_quantize_Gather_with_multiple_reduction_axes_in_8bit(mode):
```
Comment (alexsu52) [inline tests/openvino/native/quantization/test_weights_compression.py:358]: Please, add the same test for the torch backend.

**Comment by l-bat** [comment]
> Have you succeeded with accuracy validation on chatglm3 model with your PR and before it? Do you see any changes in accuracy? Could you share the results?

Model | acc | ppl
-- | -- | --
FP32 | 0.5909 | 10.2407
w/o Gather | 0.5901 | 10.2640
With Gather | 0.5892 | 10.2579

</div>
</body>
</html>

File: tests/openvino/native/quantization/test_weights_compression.py (hunk 1 above)
Comment (l-bat) [reply tests/openvino/native/quantization/test_weights_compression.py:358]: According to https://github.com/openvinotoolkit/nncf/blob/2bde5957c318997c8c81ad462deb39751a85ef9d/nncf/quantization/algorithms/weight_compression/torch_backend.py#L131-L132 `torch.nn.Embedding` can't have multiple reduction axes.

**Review by alexsu52 — APPROVED** [review]
LGTM

File: nncf/quantization/algorithms/weight_compression/weight_lowering.py [hunk 2]
```diff
@@ -121,6 +127,12 @@ def do_integer_quantization(
 
     if group_size != -1:
         # weights are reshaped from [a1, r, a2] to [a1, r//gs, gs, a2]
+        if isinstance(reduction_axis, tuple) and len(reduction_axis) == 1:
+            reduction_axis = reduction_axis[0]
+        if not isinstance(reduction_axis, int):
+            raise NotImplementedError(
```
Comment (ljaljushkin) [inline nncf/quantization/algorithms/weight_compression/weight_lowering.py:133]: what about moving this to `reshape_weight_for_grouped_quantization` ?
you won't need to do it every time before calling this method

**Review by ljaljushkin — APPROVED** [review]

**Review by AlexKoff88 — APPROVED** [review]
